core/src/main/java/com/yahoo/oak/BasicChunk.java

```diff
@@ -154,61 +602,78 @@ void release() {
     }
 
     /*----------------------- Abstract Rebalance-related Methods  --------------------------*/
+
     /**
      * Check whether this better to be rebalanced (not a necessary). Necessary rebalance is
      * triggered anyway regardless to this method.
      * */
     abstract boolean shouldRebalance();
 
     /*----------------------- Abstract Mappings-related Methods  --------------------------*/
-    /**
-     * See concrete implementation for more information
-     */
-    abstract void releaseKey(ThreadContext ctx);
-
-    /**
-     * See concrete implementation for more information
-     */
-    abstract void releaseNewValue(ThreadContext ctx);
 
     /**
      * Writes the key off-heap and allocates an entry with the reference pointing to the given key
      * See concrete implementation for more information
      */
     abstract boolean allocateEntryAndWriteKey(ThreadContext ctx, K key);
```

Comment:
`allocateEntryAndWriteKey` method can move up with all of the rest similar methods...